_extensions/odoo/layout.html

```diff
@@ -25,6 +25,13 @@
   {{ super() }}
 {%- endblock -%}
 
+{%- block linktags -%}
+{% for code, url in language_codes %}
+  <link rel="alternate" hreflang="{{ code }}" href="{{ url }}" />{% endfor %}
+  <link rel="canonical" href="{{ canonical_rel }}" />
```

Comment:
endfor should probably be on the newline [possibly using %- / -% to suppress whitespace](https://jinja.palletsprojects.com/en/2.10.x/templates/#whitespace-control), cf the block/endblock statements which make use of this feature though possibly out of cargo-culting rather than understanding)

And canonical should be un-intended to make it clear it's not part of the loop, also possibly *before* the loop?